### PR TITLE
Removed spoonacular api key from code

### PR DESCRIPTION
**Pull request by joeychin01**
Replaced plaintext api key with storing the key in datastore and requesting the data when needed.

### CODE REVIEW

**Comment by joeychin01** [comment]
I accidentally pushed to the wrong branch so this pr is extended to include my mvp changes

File: src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java [hunk 1]
```diff
@@ -34,28 +34,40 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String title = request.getParameter("dishName");
-    System.out.println(title);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query =
+      new Query("apiKey")
+        .setFilter(new Query.FilterPredicate("keyName", Query.FilterOperator.EQUAL, "spoonacular"));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    String spoonacularKey = (String) entity.getProperty("key");
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/SpoonacularNutritionServlet.java:58]: Consider wrapping this part into a helper function that can be shared by multiple components.

File: src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      new Query("apiKey")
+        .setFilter(new Query.FilterPredicate("keyName", Query.FilterOperator.EQUAL, "knowledgeGraph"));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    String knowledgeGraphKey = (String) entity.getProperty("key");
+    String firstOption = (String) request.getParameter("firstOption");
+    String secondOption = (String) request.getParameter("secondOption");
+    String firstImageURL = getImageUrl(firstOption, knowledgeGraphKey);
+    String secondImageURL = getImageUrl(secondOption, knowledgeGraphKey);
+    String imageURLs = "{\"firstURL\": \""+firstImageURL+"\", \"secondURL\": \""+secondImageURL+"\"}";
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java:63]: Consider using String.format() to replace concatinating.

File: src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    response.setContentType("application/json");
+    response.getWriter().println(imageURLs);
+  }
+
+  String getImageUrl(String dishName, String knowledgeGraphKey){
+    Client client = ClientBuilder.newClient();
+    try {
+      dishName = URLEncoder.encode(dishName);
+    }
+    catch (Exception e) {
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java:74]: Nitpick: put 'catch' after '}' on the previous line.

File: src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      System.out.println(e);
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java:75]: Should you return empty string on encoding exception?

File: src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        JSONObject searchResultResult = new JSONObject(searchResult.get("result").toString());
+        System.out.println(searchResultResult);
+        if(searchResultResult.has("image")){
+          JSONObject imageInfo = new JSONObject(searchResultResult.get("image").toString());
+          String imageURL = (String) imageInfo.get("contentUrl");
+          System.out.println(imageURL);
+          return imageURL;
+        }
+      }
+      return "https://cdn.iconscout.com/icon/premium/png-512-thumb/no-data-found-1965030-1662565.png";
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/UserChoicePictureServlet.java:94]: Define a constant.

**Pull request merged**